txcache/txCache.go

```diff
@@ -298,6 +304,30 @@ func (cache *TxCache) MaxSize() int {
 	return int(cache.config.CountThreshold)
 }
 
+// RegisterEvictionHandler registers a handler which will be called when a tx is evicted from cache
+func (cache *TxCache) RegisterEvictionHandler(handler func(hash []byte)) error {
```

Comment:
could have created a `baseCache` or a `baseEvictionHandler` instead of duplicating these functions in 2 components

Reply:
updated